local.go

```diff
@@ -39,27 +39,33 @@ func NewLocalRepository(path string) (Interface, error) {
 }
 
 // Resolve implements Interface.Resolve.
-func (r *LocalRepository) Resolve(ref *charm.Reference) (*charm.URL, error) {
+func (r *LocalRepository) Resolve(ref *charm.Reference) (*charm.URL, []string, error) {
 	if ref.Series == "" {
-		return nil, errgo.Newf("no series specified for %s", ref)
+		return nil, nil, errgo.Newf("no series specified for %s", ref)
 	}
 	u, err := ref.URL("")
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	if ref.Revision != -1 {
-		return u, nil
+		return u, nil, nil
 	}
 	if ref.Series == "bundle" {
 		// Bundles do not have revision files and the revision is not included
 		// in metadata. For this reason, local bundles always have revision 0.
-		return u.WithRevision(0), nil
+		return u.WithRevision(0), nil, nil
 	}
 	ch, err := r.Get(u)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+	// This is strictly speaking unnecessary, but just in case a bad charm is
+	// used locally, we'll check the series.
+	_, err = charm.SeriesForCharm(ref.Series, ch.Meta().Series)
+	if err != nil {
+		return nil, nil, err
 	}
-	return u.WithRevision(ch.Revision()), nil
+	return u.WithRevision(ch.Revision()), ch.Meta().Series, nil
```

Comment:
I'm not sure that we should encourage the use of charms in series-specific
directories that support multiple series.

I think it's OK to check that ch.Meta().Series is either empty or contains u.Series, but I don't think we should return the supported series here.

For example, I think that a charm that lives under $JUJU_REPOSITORY/precise/wordpress should not be able to support trusty
unless it is moved somewhere else.